### PR TITLE
PMM-10900: custom image and option to run pmm with external clickhouse

### PR DESCRIPTION
**Pull request by ritbl**
[PMM-10900](https://jira.percona.com/browse/PMM-10900)

Fully backward-compatible way to allow developers to override used Devcontainer and option to use external ClickHouse DB

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,28 +2,40 @@
 
 include Makefile.include
 
+ifeq ($(PROFILES),)
+PROFILES := 'pmm'
```
Comment (ritbl) [inline Makefile:6]: by default run as it runs now

File: docker-compose.yml [hunk 2]
```diff
@@ -19,15 +21,86 @@ services:
...
+  # PMM with external ClickHouse DB
+  ch:
+    profiles:
+      - pmm-ch
+    image: ${CH_IMAGE:-clickhouse/clickhouse-server:22.6.9.11-alpine}
+    platform: linux/amd64
+    hostname: ${CH_HOSTNAME:-ch}
+    ports:
+      - ${CH_PORT:-9000}:9000
+  pmm-managed-server-ch:
```
Comment (ritbl) [inline docker-compose.yml:78]: `ch` and `pmm-managed-server-ch` will be used when develop runs container the following way:

```
PROFILES=pmm-ch make env-up
```

File: docker-compose.yml [hunk 3]
```diff
@@ -19,15 +21,86 @@ services:
...
+      - ${PMM_PORT_CH_HTTP:-11123}:8123
+    volumes:
+      - ./:/root/go/src/github.com/percona/pmm
+      - ./Makefile.devcontainer:/root/go/src/github.com/percona/pmm/Makefile:ro # change Makefile in devcontainer
+      # caching
+      - go-modules:/root/go/pkg/mod
+      - root-cache:/root/.cache
+
+  # PMM with external ClickHouse DB
+  ch:
```
Comment (ritbl) [inline docker-compose.yml:70]: use external ClickHouse (force AMD64), as DBaaS fails (in arm64 builds, amd64 works fine) with one that is running in container.

File: managed/Makefile [hunk 4]
```diff
@@ -36,8 +36,13 @@ release:                        ## Build pmm-managed release binaries.
 	env CGO_ENABLED=0 go build -v $(PMM_LD_FLAGS) -o $(PMM_RELEASE_PATH)/ ./cmd/...
 	$(PMM_RELEASE_PATH)/pmm-managed --version
 
+ARCH=$(shell uname -m)
 release-dev:
-	go build -race -gcflags="all=-N -l" -v $(PMM_LD_FLAGS) -o $(PMM_RELEASE_PATH)/ ./cmd/...
+	@if [ $(ARCH) = "aarch64" ]; then\
+	    go build -gcflags="all=-N -l" -v $(PMM_LD_FLAGS) -o $(PMM_RELEASE_PATH)/ ./cmd/...
```
Comment (ritbl) [inline managed/Makefile:42]: to ARM do not run with -race


`-race` causes crash building pmm-managed binaries


```
# github.com/percona/pmm/managed/cmd/pmm-managed
reflect.methodValueCall: nosplit stack over 792 byte limit
reflect.methodValueCall<0>
    grows 448 bytes, calls reflect.moveMakeFuncArgPtrs<1>
        grows 288 bytes, calls internal/abi.(*IntArgRegBitmap).Get<1>
            grows 48 bytes, calls runtime.racefuncenter<1>
                grows 0 bytes, calls racefuncenter<121>
                    grows 16 bytes, calls racecall<121>
                        grows 0 bytes, calls indirect
                            grows 0 bytes, calls runtime.morestack<0>
                            8 bytes over limit
```

File: Makefile [hunk 5]
```diff
@@ -2,28 +2,40 @@
...
+endif
+
 env-up: 							## Start devcontainer.
+	COMPOSE_PROFILES=$(PROFILES) \
```
Comment (BupycHuk) [inline Makefile:10]: do we really need to set it for each item?
can we extract it outside of items?

File: .gitignore [hunk 6]
```diff
@@ -27,3 +27,5 @@ pmm-agent-dev.yaml
 
 # ViM temporary files
 *.sw[o,p]
+
+.env
```
Comment (ritbl) [inline .gitignore:31]: docker compose uses `.env` by default, it should not be committed

**Review by BupycHuk — COMMENTED** [review]
will it work if user runs just `docker-compose up`?

File: docker-compose.yml [hunk 7]
```diff
@@ -2,7 +2,9 @@
 version: '3.7'
 services:
   pmm-managed-server:
```
Comment (ritbl) [inline docker-compose.yml:4]: this section was not changed, only `profiles` section is added

File: Makefile (hunk 5 above)
Comment (ritbl) [reply Makefile:10]: I need to pass default profile (`pmm`), for some reason it does'n work with default
image

image

**Review by BupycHuk — APPROVED** [review]

**Review by tshcherban — APPROVED** [review]

**Pull request merged**